pkg/wallet/internal/mapping/mapping_list_actions.go

```
@@ -105,29 +115,40 @@ func mapListActionsAction(action wdk.WalletAction) sdk.Action {
 		Outputs:     slices.Map(action.Outputs, mapActionOutput),
 	}
 
-	return result
+	return result, nil
 }
 
 // mapActionStatus maps string status to sdk.ActionStatus
-func mapActionStatus(status string) sdk.ActionStatus {
+func mapActionStatus(status string) (sdk.ActionStatus, error) {
 	switch status {
 	case "completed":
-		return sdk.ActionStatusCompleted
+		return sdk.ActionStatusCompleted, nil
 	case "unprocessed":
-		return sdk.ActionStatusUnprocessed
+		return sdk.ActionStatusUnprocessed, nil
 	case "sending":
-		return sdk.ActionStatusSending
+		return sdk.ActionStatusSending, nil
 	case "unproven":
-		return sdk.ActionStatusUnproven
+		return sdk.ActionStatusUnproven, nil
 	case "unsigned":
-		return sdk.ActionStatusUnsigned
+		return sdk.ActionStatusUnsigned, nil
 	case "nosend":
-		return sdk.ActionStatusNoSend
+		return sdk.ActionStatusNoSend, nil
 	case "nonfinal":
-		return sdk.ActionStatusNonFinal
+		return sdk.ActionStatusNonFinal, nil
 	default:
-		return sdk.ActionStatusUnprocessed
+		return "", fmt.Errorf("unknown action status: %s", status)
+	}
+}
+
+// scriptBytes converts a hex string to script bytes, returns nil if conversion fails
+func scriptBytes(hexString string) []byte {
+	if hexString == "" {
+		return nil
+	}
+	if script, err := script.NewFromHex(hexString); err == nil {
```

please, return the error also if there is one.

---

As @dorzepowski  mentioned in the previous comment, we should return the missing error.